store/consul/consul.go

```diff
@@ -467,6 +467,13 @@ func (s *Consul) AtomicDelete(key string, previous *store.KVPair) (bool, error)
 	}
 
 	p := &api.KVPair{Key: s.normalize(key), ModifyIndex: previous.LastIndex}
+
+	// Extra Get operation to check on the key
+	_, err := s.Get(key)
+	if err != nil && err == store.ErrKeyNotFound {
+		return false, err
+	}
+
```

Comment:
There is no way you can do that with the Consul client. It just bubbles up the http error (not typed like etcd or zookeeper) so you need to check for the error string in the `DeleteCAS` call (which is really bad) or infer this information from the return arguments.

I prefer doing the extra call to Get and assuming that if pair is nil, the key does not exists. It does not cost too much as `AtomicDelete` should be a "rare" operation anyway.